charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml

```diff
@@ -33,7 +33,35 @@ groups:
   # Recording rules for container count per container
   - record: seed:kube_pod_container_info:count_by_container
     expr: count(kube_pod_container_info) by (container)
-
+  
+  # Recording rules for container restart count per container
+  - record: seed:kube_pod_container_status_restarts_total:sum_by_container
+    expr: sum(kube_pod_container_status_restarts_total) by (container)
+  
+  # Recording rules for deployment spec replicas per deployment
+  - record: seed:kube_deployment_spec_replicas:sum_by_deployment
+    expr: sum(kube_deployment_spec_replicas) by (deployment)
+  
+  # Recording rules for deployment status replicas per deployment
+  - record: seed:kube_deployment_status_replicas:sum_by_deployment
+    expr: sum(kube_deployment_status_replicas) by (deployment)
+  
+  # Recording rules for deployment status replicas available per deployment
+  - record: seed:kube_deployment_status_replicas_available:sum_by_deployment
```

Comment:
The metrics look fine, but could you give some motivation as to why we need the these metrics? Specifically seed:kube_deployment_status_replicas:sum_by_deployment and seed:kube_deployment_status_replicas_available:sum_by_deployment. Is this to correlate expected state to actual state?

Reply:
> Specifically seed:kube_deployment_status_replicas:sum_by_deployment and seed:kube_deployment_status_replicas_available:sum_by_deployment. Is this to correlate expected state to actual state?

Yes. That and to make sure the right individual components overall are not having availability/downtime issues. Helps in seeing if say `kube-apiserver`s are generally having availability/downtime issues after some scaling (HPA,VPA/HVPA) optimization.

> The metrics look fine, but could you give some motivation as to why we need the these metrics?

Where would you like such explanation? In the comments or in the PR's release notes?

Reply:
> Where would you like such explanation? In the comments or in the PR's release notes?

This is fine, I just wanted to know the motivation of having these metrics. Thanks!